test/test_processing_utils.py

```diff
@@ -54,12 +54,12 @@ def test_encode_pil_to_base64(self):
         output_base64 = processing_utils.encode_pil_to_base64(img)
         assert output_base64 == deepcopy(media_data.ARRAY_TO_BASE64_IMAGE)
 
-    def test_save_pil_to_file_keeps_pnginfo(self):
+    def test_save_pil_to_file_keeps_pnginfo(self, tmp_path):
         input_img = Image.open("gradio/test_data/test_image.png")
         input_img = input_img.convert("RGB")
         input_img.info = {"key1": "value1", "key2": "value2"}
 
-        file_obj = processing_utils.save_pil_to_file(input_img)
+        file_obj = components.Image().pil_to_temp_file(input_img, dir=tmp_path)
```

Comment:
I don't see why these were moved from `processing_utils` to where they need an instance for no reason? Maybe `image_utils`? `file_utils`?

Reply:
It's to keep all of the functions that create temporary files in one place (specifically in the `IOComponent` class)